app/services/forest_liana/operator_date_interval_parser.rb

```diff
@@ -1,114 +1,94 @@
 module ForestLiana
   class OperatorDateIntervalParser
     PERIODS = {
-      :$yesterday => { duration: 1, period: 'day' },
-      :$previousWeek => { duration: 1, period: 'week' },
-      :$previousMonth => { duration: 1, period: 'month' },
-      :$previousQuarter => { duration: 3, period: 'month',
-                            period_of_time: 'quarter' },
-      :$previousYear => { duration: 1, period: 'year' },
-      :$weekToDate => { duration: 1, period: 'week', to_date: true },
-      :$monthToDate => { duration: 1, period: 'month', to_date: true },
-      :$quarterToDate => { duration: 3, period: 'month',
-                          period_of_time: 'quarter', to_date: true },
-      :$yearToDate => { duration: 1, period: 'year', to_date: true }
+      :yesterday => { duration: 1, period: 'day' },
+      :previous_week => { duration: 1, period: 'week' },
+      :previous_week_to_date => { duration: 1, period: 'week', to_date: true },
+      :previous_month => { duration: 1, period: 'month' },
+      :previous_month_to_date => { duration: 1, period: 'month', to_date: true },
+      :previous_quarter => { duration: 3, period: 'month', period_of_time: 'quarter' },
+      :previous_quarter_to_date => { duration: 3, period: 'month', period_of_time: 'quarter', to_date: true },
+      :previous_year => { duration: 1, period: 'year' },
+      :previous_year_to_date => { duration: 1, period: 'year', to_date: true }
     }
 
-    PERIODS_PAST = '$past';
-    PERIODS_FUTURE = '$future';
-    PERIODS_TODAY = '$today';
-
-    PERIODS_PREVIOUS_X_DAYS = /^\$previous(\d+)Days$/;
-    PERIODS_X_DAYS_TO_DATE = /^\$(\d+)DaysToDate$/;
-    PERIODS_X_HOURS_BEFORE = /^\$(\d+)HoursBefore$/;
-    PERIODS_X_HOURS_AFTER = /^\$(\d+)HoursAfter$/;
-
-    def initialize(value, timezone)
-      @value = value
+    OPERATOR_PAST = 'past';
+    OPERATOR_FUTURE = 'future';
+    OPERATOR_TODAY = 'today';
+
+    OPERATOR_PREVIOUS_X_DAYS = 'previous_x_days'
```

Comment:
If I specify a big number it crash 
![Capture d’écran 2019-07-31 à 10 47 57](https://user-images.githubusercontent.com/3263749/62198826-a9f46f80-b382-11e9-947e-47ec97976c71.png)

Reply:
Worked before, not since and I can't find why (might need assistance on that)